lib/hedge_fund_interview/prisms/utils.ex

```diff
@@ -1,8 +1,12 @@
 defmodule HedgeFundInterview.Prisms.Utils do
   def send_signal(signal) do
     endpoint = System.get_env("SPECTRAL_API_AGENTS_ENDPOINT")
+    api_key = System.get_env("SPECTRAL_API_KEY")
 
-    case Req.post("#{endpoint}/signals/register", json: signal) do
+    case Req.post("#{endpoint}/signals/register",
+           json: signal,
+           headers: [{"authorization", "Bearer #{api_key}"}]
```

Comment:
@Ian-GL there is a more standard way to do this with Req. Please check my PR in this repo (or the Req docs).